### ISSUE TITLE
Not compiling on Windows from Travis-CI and Azure DevOps Pipeline.

### ISSUE THREAD

**Issue by SeanOMik** [machinezone/IXWebSocket#242]
I set up a CI on Azure DevOps to compile my project and after first learning how it works, I eventually got something to attempt to compile but it failed with some weird errors:
```D:/a/1/s/thirdparty/IXWebSocket/ixwebsocket/IXDNSLookup.cpp: In static member function 'static addrinfo* ix::DNSLookup::getAddrInfo(const string&, int, std::__cxx11::string&)':
D:/a/1/s/thirdparty/IXWebSocket/ixwebsocket/IXDNSLookup.cpp:47:26: error: 'AI_ADDRCONFIG' was not declared in this scope
         hints.ai_flags = AI_ADDRCONFIG | AI_NUMERICSERV;
                          ^~~~~~~~~~~~~
D:/a/1/s/thirdparty/IXWebSocket/ixwebsocket/IXDNSLookup.cpp:47:26: note: suggested alternative: 'LUP_ADDRCONFIG'
         hints.ai_flags = AI_ADDRCONFIG | AI_NUMERICSERV;
                          ^~~~~~~~~~~~~
                          LUP_ADDRCONFIG
D:/a/1/s/thirdparty/IXWebSocket/ixwebsocket/IXDNSLookup.cpp:47:42: error: 'AI_NUMERICSERV' was not declared in this scope
         hints.ai_flags = AI_ADDRCONFIG | AI_NUMERICSERV;
                                          ^~~~~~~~~~~~~~
D:/a/1/s/thirdparty/IXWebSocket/ixwebsocket/IXDNSLookup.cpp:47:42: note: suggested alternative: 'NI_NUMERICSERV'
         hints.ai_flags = AI_ADDRCONFIG | AI_NUMERICSERV;
                                          ^~~~~~~~~~~~~~
                                          NI_NUMERICSERV
```
Using my IDE on my PC I saw that these things are defined in `windef2.h`, this made me believe that it may be something to do with Azure's Windows image so I decided to try to set it up on Travis-CI. I got that setup fairly quickly but ran into the same issues. Here's some files you would probably want to take a look at: 
* [.travis.yml](https://github.com/DisCPP/DisCPP/blob/3254f90673d77d26c1a75f3ba0550e5a62895e49/.travis.yml)
* [azure-pipelines.yml](https://github.com/DisCPP/DisCPP/blob/3254f90673d77d26c1a75f3ba0550e5a62895e49/azure-pipelines.yml)
* [CMakeLists.txt](https://github.com/DisCPP/DisCPP/blob/3254f90673d77d26c1a75f3ba0550e5a62895e49/CMakeLists.txt)
* [Azure Pipeline Output](https://dev.azure.com/seanomik/DisCPP/_build/results?buildId=470&view=logs&j=5d72d3a4-37a1-5685-94fb-38a4feb625a0&t=62f00f4f-a09b-595a-13a2-ebd1b705ca5b)
* [Travis-CI Output](https://travis-ci.org/github/DisCPP/DisCPP/jobs/723267757#L488)

My compiler on Azure, and Travis-CI is GCC 8.1.0. Everything compiles fine on Linux.
If you have any questions ask them, and I'll ask as fast as I can. Thanks!

**Comment by bsergean** [comment]
We could try to have a GitHub action that builds on Windows with gcc.

https://docs.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners <https://docs.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners>

https://docs.github.com/en/actions/reference/software-installed-on-github-hosted-runners <https://docs.github.com/en/actions/reference/software-installed-on-github-hosted-runners>

Those are the machines available.

Chocolatey 0.10.15 is available, maybe we could use it to fetch gcc. I see you are doing something similar with travis.

One thing I know is that mingw compiler is not supported as it requires to not std::threads or weird macro.

This thread is very old but talks about AI_ADDRCONFIG

https://sourceforge.net/p/libmms/bugs/10/

**Comment by SeanOMik** [comment]
I think chocolatey has gcc 8.1.0 installed by default, which Travis and Azure is using. I'll take a look if I can install a newer version. I was able to get it working by using Clang 10.0. After looking up where gcc is coming from via Chocolatey it looks like it does come from the MinGW compiler. I'll just tell my users that GCC is supported on windows. I don't think its a popular compiler on Windows anyway.

**Comment by bsergean** [comment]
Sounds good.

I can try to setup a clang 10 CI on GitHub actions, but if you can help with that it would be awesome. I can look at your travis files too if you have one with clang 10.

I think that clang has some support from google who must use it to build the Chrome browser. Not sure about gcc on windows those days.

**Comment by bsergean** [comment]
This is how I do CI on windows

https://github.com/machinezone/IXWebSocket/blob/master/.github/workflows/unittest_windows.yml

**Comment by SeanOMik** [comment]
I tried to setup a clang CI on my project but for some reason it just completely ignores me setting the CXX compiler and uses MSVC anyway. 

Here's my action yaml:
https://github.com/DisCPP/DisCPP/blob/6501f0bc4ade6e9abb96e89526c572c8e36e1bae/.github/workflows/unittest_compile.yml

**Comment by SeanOMik** [comment]
It would be pretty cool to get WinSSL working. And I do agree that CI is always a pain for me to get setup and it baffles me when I see people get it setup first try. 

I tried a bunch of different things for GitHub actions for clang. I actually eventually got it working on [Azure](https://dev.azure.com/seanomik/DisCPP/_build/results?buildId=501&view=results). I'll mess around a little bit with GitHub actions and I'll create a pull request if I get it working.
